osu.Game/Rulesets/UI/DrawableRuleset.cs

```diff
@@ -60,6 +60,13 @@ public abstract class DrawableRuleset<TObject> : DrawableRuleset, IProvideCursor
         /// </summary>
         public override Playfield Playfield => playfield.Value;
 
+        private readonly Lazy<PlayfieldAdjustmentContainer> playfieldAdjustmentContainer;
+
+        /// <summary>
+        /// The playfield adjustment container.
+        /// </summary>
+        public virtual PlayfieldAdjustmentContainer PlayfieldAdjustmentContainer => playfieldAdjustmentContainer.Value;
```

Comment:
Unsure why `virtual`. Don't really believe this should be overridable, `CreatePlayfieldAdjustmentContainer()` should suffice for a customisation API surface.